### PR TITLE
two functions to check crystal equivalence

### PR DESCRIPTION
**Pull request by cpignedoli**
to use them:
formula=ase_structure.get_chemical_formula()
structures_with_same_formula=formula_in_qb(formula)
for one in structures_with_same_formula:
    if check_crystal_equivalence(ase_structure,one[0]):
        print("find equivalent structure with PK",one[1])

### CODE REVIEW

File: aiidalab_widgets_base/search_qb.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from __future__ import print_function
+from collections import OrderedDict
+from IPython.display import display, clear_output, HTML
```
Comment (ltalirz) [inline aiidalab_widgets_base/search_qb.py]: Perhaps check a few of these imports, most won't be necessary

File: aiidalab_widgets_base/crystal_sim_crystal.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+from ase.data import atomic_numbers, atomic_names
+from ase.spacegroup import Spacegroup
+from numpy.linalg import norm
+
+
+def ase_to_spgcell(ase_atoms):
+    return (ase_atoms.get_cell(),
+            ase_atoms.get_scaled_positions(),
+            ase_atoms.get_atomic_numbers())
+def a_equiv_b(a,b):
```
Comment (ltalirz) [inline aiidalab_widgets_base/crystal_sim_crystal.py]: perhaps a more expressive  name might be "check_crystal_equivalence(crystal_a, crystal_b)"

**Comment by yakutovicha** [comment]
@cpignedoli , you said you have and idea for the update of this `a_equiv_b`. Should we wait until this is done?

File: aiidalab_widgets_base/crystal_sim_crystal.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            sorted_b = np.array(sorted([ list(row) for row in list(fullb) ]))
+            if np.allclose(sorted_n, sorted_b, atol=1e-5):
+                return True
+        return False
+
+    if samecell and samenatoms and samespg:
+        cell = cryst_a['std_lattice']
+        rng1 = range(1, int(norm(cell[0])/2.))
+        rng2 = range(1, int(norm(cell[1])/2.))
+        rng3 = range(1, int(norm(cell[2])/2.))
```
Comment (yakutovicha) [inline aiidalab_widgets_base/crystal_sim_crystal.py]: Please leave a comment here saying why you are dividing by 2. Like saying that cell vectors can't be smaller than 2.

**Comment by cpignedoli** [comment]
> @cpignedoli , you said you have and idea for the update of this `a_equiv_b`. Should we wait until this is done?

No it was not here, i do not know how this function can be made faster my comment was for "structures_multi.py": there, instead of calling formula_in_qb, one should probably create a list of all chemical formulas present in the database then check the PKs that hve a given formula....

**Comment by cpignedoli** [comment]
I implemented your suggestions, please check if we can now merge.

**Comment by yakutovicha** [comment]
Thanks @cpignedoli

**Pull request merged**